Sources/Sentry/SentryScope.m

```diff
@@ -456,16 +447,30 @@ - (SentryEvent *__nullable)applyToEvent:(SentryEvent *)event
         event.level = level;
     }
 
-    id<SentrySpan> span;
-    @synchronized(_spanLock) {
-        span = self.span;
+    NSMutableDictionary *newContext;
+    if (nil == event.context) {
+        newContext = [self context].mutableCopy;
+    } else {
+        newContext = [NSMutableDictionary new];
+        [newContext addEntriesFromDictionary:[self context]];
+        [newContext addEntriesFromDictionary:event.context];
     }
 
-    if (![event.type isEqualToString:SentryEnvelopeItemTypeTransaction] &&
-        [span isKindOfClass:[SentryTracer class]]) {
-        event.transaction = [(SentryTracer *)span name];
+    if (self.span != nil) {
+        id<SentrySpan> span;
+        @synchronized(_spanLock) {
+            span = self.span;
```

Comment:
span could be `nil` now since the check is done outside the guard.
It requires double checking now: